src/mmc_windows.c

```diff
@@ -33,163 +33,162 @@ void mmc_finalize()
-        device->size = geometry.DiskSize.QuadPart;
-        return 1;
-    } else {
-        return 0;
-    }
-}
-
-/**
- * @brief Scan for SDCards and other removable media
- * @param devices where to store detected devices and some metadata
- * @param max_devices the max to return
- * @return the number of devices found
- */
-int mmc_scan_for_devices(struct mmc_device *devices, int max_devices)
-{
-    DWORD  Error                = ERROR_SUCCESS;
-    HANDLE FindHandle           = INVALID_HANDLE_VALUE;
-    size_t Index                = 0;
-    BOOL   Success              = FALSE;
-    WCHAR  VolumeName[MAX_PATH] = L"";
-    int device_count = 0;
-
-    //  Enumerate all volumes in the system.
-    FindHandle = FindFirstVolumeW(VolumeName, ARRAYSIZE(VolumeName));
-    if (FindHandle == INVALID_HANDLE_VALUE) {
-        Error = GetLastError();
-        fprintf(stderr, "FindFirstVolumeW failed with error code %d\n", Error);
+    if (!status || extents.NumberOfDiskExtents > 1) {
+        // Ignore it if we can't query its extents or it is backed by
+        // more than one extent because it's probably not an SD card.
         return 0;
     }
 
-    while (device_count < max_devices) {
-        //  Skip the \\?\ prefix and remove the trailing backslash.
-        Index = wcslen(VolumeName) - 1;
-
-        if (VolumeName[0]     != L'\\' ||
-            VolumeName[1]     != L'\\' ||
-            VolumeName[2]     != L'?'  ||
-            VolumeName[3]     != L'\\' ||
-            VolumeName[Index] != L'\\') {
-            Error = ERROR_BAD_PATHNAME;
-            break;
-        }
-
-        WCHAR volume_path[Index + 1];
-        memcpy(volume_path, VolumeName, Index * sizeof(WCHAR));
-        volume_path[Index] = L'\0';
-
-        if (mmc_query_volume_path(volume_path, &devices[device_count]))
-            device_count++;
-
-        //
-        //  Move on to the next volume.
-        Success = FindNextVolumeW(FindHandle, VolumeName, ARRAYSIZE(VolumeName));
-
-        if ( !Success )
-        {
-            Error = GetLastError();
-
-            if (Error != ERROR_NO_MORE_FILES)
-            {
-                fprintf(stderr, "FindNextVolumeW failed with error code %d\n", Error);
-                break;
-            }
-
-            //  Finished iterating
-            //  through all the volumes.
-            Error = ERROR_SUCCESS;
-            break;
-        }
-    }
-
-    FindVolumeClose(FindHandle);
-    return device_count;
+    return extents.Extents[0].DiskNumber;
 }
 
-int mmc_umount_all(const char *mmc_device)
-{
-    WCHAR  VolumeName[MAX_PATH];
-    MultiByteToWideChar(CP_UTF8, 0, mmc_device, 0, VolumeName, MAX_PATH);
-
-    HANDLE volume_handle = CreateFile(VolumeName,
+/*
+ * @brief Attempt to dismount the specified volume, exit with failure message if unsuccessful
+ * @param volume_name the name of the volume as a Wide String
+ */
+void dismount_volume(LPWSTR volume_name) {
```

Comment:
Use static for `dismount_volume`. Also, is `dismount` the Windows terminology? fwup uses `unmount` everywhere else, and it would be nice to be consistent unless it would confuse Windows users.